frontend/pages/hosts/ManageHostsPage/ManageHostsPage.tsx

```diff
@@ -459,9 +488,27 @@ const ManageHostsPage = ({
     retrieveHostCount(options);
   };
 
+  let teamSync = false;
+  if (currentUser && availableTeams) {
+    // team_id starts as a string and needs to be parsed. If the user provides a non-numeric
+    // string, it will parse as NaN. However, we need to handle if team_id is undefined (which is
+    // would be the case for the root path to all teams) because undefined also parses to NaN.
+    // We want to be able to check that teamIdParam equals undefined in the `else if` below because
+    // that will correspond to the undefined state `currentTeam`, which represents the root path.
+    const teamIdParam = queryParams.team_id
+      ? parseInt(queryParams.team_id, 10)
+      : undefined;
+    if (currentTeam?.id && !teamIdParam) {
+      teamSync = true;
+    } else if (teamIdParam === currentTeam?.id) {
+      teamSync = true;
+    }
+  }
+
+  // TODO: Exit early if !availableTeams?
```

Comment:
Could this be documented somewhere else? Trying to move away from TODOs as comments.